### ISSUE TITLE
ExactQuotientFailed error when using 1.0 instead of 1 in a complex integral with cosines

### ISSUE THREAD

**Issue by rupof** [sympy/sympy#27189]
Hi!

I found the following bug:

Doing, 
```
import sympy as sp
x = sp.Symbol("x")
sp.integrate(1*sp.sin(x)*sp.cos(x)**2*sp.I, x,)
```
Returns

`-I*exp(3*I*x0)/24 - I*exp(I*x0)/8 - I*exp(-I*x0)/8 - I*exp(-3*I*x0)/24
`
On the other hand, by using 1.0 instead of 1:

```
x = sp.Symbol("x")
sp.integrate(1.0*sp.sin(x)*sp.cos(x)**2*sp.I, x,)
```

The following errors are output:

```
ExactQuotientFailed: [(1.0 + 0.0j), (0.0 + 0.0j), (0.0 + 0.0j)] does not divide [(1.0 + 0.0j)]
During handling of the above exception, another exception occurred:
ExactQuotientFailed: 1.0*_t0**2 does not divide 1.00000000000000
```

Curiously, the error only appears for sin(x)^q cos(x)^p, with p > 2 and q=1 or p=1 and q=2. For other combinations, it works well

This is my first time contributing. I am happy to try to help, if someone can point me in the right direction.

Thank you very much!

**Comment by oscarbenjamin** [comment]
I reproduce this problem with SymPy 1.12 but I believe this was fixed as of SymPy 1.13.

I just bisected the fix to b5eb8531769e345dea4ae432c7fe51723191991e from gh-25982. The effect of that PR is that somewhere in the polys code the GCD of two polynomials in `CC[x]` is computed in `QQ_I[x]`. This means that the complex floats are converted to complex numbers with exact rational coefficients to avoid using the PRS algorithm with floating point coefficients which otherwise suffers from this exact quotient problem.
